packages/replayio/src/utils/recordings/upload/uploadRecording.ts

```diff
@@ -198,54 +194,115 @@ async function uploadRecordingFileInParts({
   partLinks: string[];
   recordingPath: string;
 }): Promise<string[]> {
-  const { size: totalSize } = statSync(recordingPath);
-
-  const results = await promiseMap<string, string>(
-    partLinks,
-    async (url: string, index: number) => {
-      return retryWithLinearBackoff(
-        async () => {
-          const partNumber = index + 1;
-          const start = index * chunkSize;
-          const end = Math.min(start + chunkSize, totalSize) - 1; // -1 because end is inclusive
-
-          debug("Uploading part %o", {
-            partNumber,
-            start,
-            end,
-            totalSize,
-            chunkSize,
-          });
-          return uploadPart(url, { recordingPath, start, end }, end - start + 1);
-        },
-        error => {
-          debug(`Failed to upload part ${index + 1}. Will be retried: %o`, error);
-        },
-        10
-      );
-    },
-    { concurrency: 10 }
-  );
+  const { size: totalSize } = await stat(recordingPath);
+  const abortController = new AbortController();
+
+  const partsUploads = partLinks.map((url: string, index: number) => {
```

Comment:
During my testing, if I surge with all chunks at once, most of them will timeout with a bad connection. It will either end up in more retries or complete failure (hitting max retry limit). I think we should retain concurrency control.

Reply:
This is still capped as every scheduled chunk gets put on the queue. This line here just awaits the results - but those promises are not executed eagerly (they have concurrency set to 10).

Reply:
Or are you saying that you ended up with this situation when testing this PR locally? 🤔

Reply:
No, I haven't checked out locally. I looked PR only superficially, for that matter. I didn't want to block PR just because I'm occupied with other stuff.